pkg/cmd/comp.go

```diff
@@ -90,7 +91,15 @@ func NewCompRunCommands(c types.Args, ioStreams util.IOStreams) *cobra.Command {
 func GetWorkloadNameFromArgs(args []string) (string, error) {
 	argsLength := len(args)
 	if argsLength < 1 {
-		return "", errors.New("must specify name for workload")
+		workloadList,err := InstalledWorkloads()
+		if err != nil {
+			return "",err
+		}
+		errMsg := "can not find workload, check workloads by `vela workloads` and choose a suitable one."
+		if workloadList != "" {
+			errMsg = fmt.Sprintf("must specify name for workload ,please choose from %s.",workloadList)
+		}
+		return "", errors.New(errMsg)
```

Comment:
it's infact component name, not workload name, changing the error message is fine

Reply:
Change it to "must specify workload, please choose from 'deployment','containerized'" ?

Reply:
No, should change to "must specify name for component"

Reply:
OK

Reply:
Maybe there's still misunderstanding here, vela comp run command is like below:

```
vela comp run -t <workload_type> component-name
```

So the lengh of arguments less than 1 just means user forgot to specify component name here, but your code is telling user to specify workload type which should be specify by `-t` parameter.

@sunny0826 @zzxwill

Reply:
I get it, my bad. Maybe I need to regroup my thoughts.